showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITLongRunningOperation.java

```diff
@@ -39,6 +43,13 @@
  */
 public class ITLongRunningOperation {
```

Comment:
Are we sure that this already-complex integration test the best place to add these new assertions? Rephrased: Is the showcase server needed to verify the client's logging behavior?

The changes we're making are in `com.google.api.gax.longrunning.OperationTimedPollAlgorithm`, but we don't yet have any unit tests for this class. Is it possible for us to write a direct unit test that focuses purely on the logging behavior of `OperationTimedPollAlgorithm.shouldRetry()`?

Without looking at this PR and only the logic we're trying to test, I would have expected to see a set of unit tests that were roughly:
```
// WARNING: Complete pseudocode below...
public void testTroubleshootingLinkProvided() {
  LOGGER.addHandler(handler);
  assertThrows(..., algorithm.shouldRetry(NEXT_ATTEMPT_SHOULD_NOT_RETRY));
  assertThat(handler.getLogMessage().exists())
  assertThat(handler.getLogMessage().contains(LRO_TROUBLESHOOTING_LINK));
}
public void testTroubleshootingLinkNotProvided() {
  LOGGER.addHandler(handler);
  assertThat(algorithm.shouldRetry(NEXT_ATTEMPT_SHOULD_RETRY), is(true));
  assertThat(handler.getLogMessage().doesNotExist())
}
```

Then the integration test can be kept clear of client-side-only implementation details, allowing the person debugging a problem to focus only on the relevant interactions.

Reply:
@burkedavison thank you. At some point writing the test in this file helped me find that the exception thrown in `OperationTimedPollAlgorithm` does not bubble up to the client (hence the use of a logger), but maybe that's just part of finding out about the behavior of LROs. I will move the test case to a unit test in gax